tests/Tests/QuickCheckUtils.hs

```diff
@@ -76,54 +79,68 @@ instance Arbitrary BL.ByteString where
 -- | For tests that have O(n^2) running times or input sizes, resize
 -- their inputs to the square root of the originals.
 newtype Sqrt a = Sqrt { unSqrt :: a }
-    deriving (Eq, Show)
+ deriving (Eq, Show)
 
 instance Arbitrary a => Arbitrary (Sqrt a) where
-    arbitrary = fmap Sqrt $ sized $ \n -> resize (smallish n) arbitrary
-        where
-            smallish = round . (sqrt :: Double -> Double) . fromIntegral . abs
-    shrink = map Sqrt . shrink . unSqrt
+    arbitrary = coerce $ sized $ \n -> resize (smallish n) $ arbitrary @a
+     where
+      smallish = round . (sqrt :: Double -> Double) . fromIntegral . abs
+    shrink = map coerce . shrink @a . coerce
```

Comment:
`coerce (shrink @a)` should work as well.

Reply:
@phadej you use `~=` as a strong equivalence, strong equality? I am interested in types of equalities, and you are a good person to ask how `~=` symbol is defined/used by Haskellers.

Reply:
I use it as some vague equivalence. After expanding `.`. the `coerce . f = (\x -> coerce (f x))` which is different then `coerce f` (e.g. if `f = undefined` and you `seq` it). But here it shouldn't matter.

Reply:
Ok, nice to get details.

Because math denotation of equalities/equivalences is sadly a mess. I know as I have interest & tried to consolidate the denotation/knowledge on equality/equivalences in the math fields. In some lectures, seen the usage of `≈` as a strong equivalence, which completely puzzled my protocol, so I ask.

I'd used:
`~` is a weak similarity.
`≈` is an isomorphism, "up to something" default, "up to isomorphism", equivalence.
`≃` is a strong iso.
`=` is a "default" equality.
`≅` is stronger than external equality.
`≡` is strong equality (when one wants to say that it is literally the one thing).

So - the way you use it.

And so far not learned to remember & think through bottoms when thinking about equalities of the code, so also as others love to simplify Hask.